src/fu-engine.c

```diff
@@ -2744,6 +2745,21 @@ fu_engine_add_releases_for_device_component (FuEngine *self,
 		if (checksums->len == 0)
 			continue;
 
+		/* test for upgrade or downgrade */
+		vercmp = fu_common_vercmp (fwupd_release_get_version (rel),
+					  fu_device_get_version (device));
+		if (vercmp > 0)
+			fwupd_release_add_flag (rel, FWUPD_RELEASE_FLAG_IS_UPGRADE);
+		else if (vercmp < 0)
+			fwupd_release_add_flag (rel, FWUPD_RELEASE_FLAG_IS_DOWNGRADE);
+
+		/* lower than allowed to downgrade to */
+		if (fu_device_get_version_lowest (device) != NULL &&
+		    fu_common_vercmp (fwupd_release_get_version (rel),
+				      fu_device_get_version_lowest (device)) < 0) {
```

Comment:
This looks like a logic change - wasn't it `<=` before?

Reply:
I'm impressed you noticed that! I wasn't sure. I figured if you had version-lowest=1.2.3 and you had 1.2.3 installed, and 1.2.3 in the repos then you should be able to "install" it, unlike before where it would be blocked.

Reply:
I guess that does match more closely how the ESRT defines `Lowest Supported Firmware Version` too.  OK.